### ISSUE TITLE
no underline any more for open files

### ISSUE THREAD

**Issue by michaeldcwolf** [oviradoi/npp-explorer-plugin#33]
The highlighting (underline) functionality is not working any more for open files.
Windows11 and 32bit notepad++

**Comment by michaeldcwolf** [comment]
![Screenshot 2022-09-08 114946](https://user-images.githubusercontent.com/26809426/189092120-c686c740-67b8-4068-bb5e-7600aacb622e.jpg)

**Comment by oviradoi** [comment]
Thanks for reporting this bug. This was fixed in version 1.9.6.

This was caused by Notepad++'s new dark mode support. It would override the drawing logic inside the Explorer plugin in order to apply the new dark theme. As a result, the code that would draw the underline was not run anymore.

In order to fix this I had to disable Notepad++'s theming and implement dark mode inside the Explorer plugin. As a result, it will look slightly different in dark mode, but the underline will work again now.

**Comment by michaeldcwolf** [comment]
Underline is not working again :-)

**Comment by oviradoi** [comment]
Hmm, I have not changed anything for a long time. Can you tell me what version of notepad++, the version of the extension and the version of Windows you are using?

A screenshot can also be helpful.

**Comment by michaeldcwolf** [comment]
Hi,

 

I use v8.6.2 64bit notepad++ and Explorer 1.9.9.

Same effect on Windows11 and Window10. Seems to be introduced this year.

 

Best Regards, Michael

 

Von: Ovidiu Rădoi ***@***.***> 
Gesendet: Dienstag, 19. März 2024 15:06
An: oviradoi/npp-explorer-plugin ***@***.***>
Cc: Michael Wolf (SSI Schäfer) ***@***.***>; Author ***@***.***>
Betreff: Re: [oviradoi/npp-explorer-plugin] no underline any more for open files (Issue #33)

 

Hmm, I have not changed anything for a long time. Can you tell me what version of notepad++, the version of the extension and the version of Windows you are using?

A screenshot can also be helpful.

**Comment by oviradoi** [comment]
I'm afraid I cannot reproduce the issue. Please attach a screenshot or a gif so maybe I can get more clues about it.

**Comment by michaeldcwolf** [comment]
Von: Ovidiu Rădoi ***@***.***> 
Gesendet: Samstag, 30. März 2024 07:19
An: oviradoi/npp-explorer-plugin ***@***.***>
Cc: Michael Wolf (SSI Schäfer) ***@***.***>; Author ***@***.***>
Betreff: Re: [oviradoi/npp-explorer-plugin] no underline any more for open files (Issue #33)

 

I'm afraid I cannot reproduce the issue. Please attach a screenshot or a gif so maybe I can get more clues about it.

**Comment by oviradoi** [comment]
I think you are trying to attach screenshots via email replies, but the screenshots don't show up in the github issue. Please reply directly on github and attach the screenshot there. Thanks.

**Comment by michaeldcwolf** [comment]
Sorry for the delay. But you can have a look at the old screenshot above. Situation is the same.

**Comment by michaeldcwolf** [comment]
![Screenshot 2024-04-30 084229](https://github.com/oviradoi/npp-explorer-plugin/assets/26809426/c3768f10-044d-40fa-8c10-d79b17f701b0)

**Comment by oviradoi** [comment]
Now I see it's in the Favorites panel.

Because of a change in Notepad++, plugins are no longer able to customize the way they draw items in TreeView controls, leading to this issue.

I have reported this to Notepad++, and I created a fix for this issue over there.

The fix has been merged and it should be included in the next release of Notepad++. Then the underlining should work again.

I will update the issue when the release will be made.
